### PR TITLE
Remove ACE_HAS_DLSYM_SEGFAULT_ON_INVALID_HANDLE in config-linux.h

### PR DESCRIPTION
**Pull request by likema**
In Linux, it is valid to `dlsym` with `RTLD_DEFAULT` that is equal to
`ACE_SHLIB_INVALID_HANDLE`.

### CODE REVIEW

**Comment by jwillemsen** [comment]
Can you extend one of the ACE unit tests to call ACE_OS::dlsym with an invalid handle so that we know it doesn't segfault?

**Comment by likema** [comment]
Added test case into DLL_Test

File: ACE/tests/DLL_Test.cpp [hunk 1]
```diff
@@ -168,6 +168,11 @@ run_main (int, ACE_TCHAR *[])
               ACE_TEXT ("Dynamically Linkable Libraries not supported on this platform\n")));
 #endif /* ACE_HAS_DYNAMIC_LINKING */
 
+#if defined(ACE_LINUX)
+  ACE_TEST_ASSERT (RTLD_DEFAULT == ACE_SHLIB_INVALID_HANDLE);
+  ACE_TEST_ASSERT (ACE_OS::dlsym (ACE_SHLIB_INVALID_HANDLE, "open"));
```
Comment (jwillemsen) [inline ACE/tests/DLL_Test.cpp]: Why not test this on all platforms?

Comment (likema) [reply ACE/tests/DLL_Test.cpp]: Fixed

**Pull request merged**